include/ilang/ila/instr_lvl_abs.h

```diff
@@ -291,6 +334,8 @@ class InstrLvlAbs : public Object,
   VarMap inputs_;
   /// The set of state variables.
   VarMap states_;
+  /// The set of known objects.
+  ContainerMap objects_;
```

Comment:
An object can contain both input and state variables -- I don't require that the object has to be only one or the other, and even sometimes mix the two types in order to build a vector of data-atoms. That's why I'm trying to avoid encouraging users to think of objects as states or inputs; they should see them as collections of states and inputs. Is there a more clear way to convey this idea?

Reply:
At the same time, for clarity, I only gave the user functionality to create entirely state or entirely input objects, which they can later separate and combine as they please.